src/Workspaces/Core/Portable/DesignerAttribute/DesignerInfo.cs

```diff
@@ -0,0 +1,65 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+#nullable enable
+
+using System;
+using Microsoft.CodeAnalysis.ErrorReporting;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis.DesignerAttribute
+{
+    internal struct DesignerInfo
+    {
+        /// <summary>
+        /// Our current serialization format.  Increment whenever it changes so that we don't read
+        /// bogus data when reading older persisted data.
+        /// </summary>
+        private const string SerializationFormat = "1";
+
+        /// <summary>
+        /// The category specified in a <c>[DesignerCategory("...")]</c> attribute.
+        /// </summary>
+        public string? Category;
+
+        /// <summary>
+        /// The document this <see cref="Category"/> applies to.
+        /// </summary>
+        public DocumentId DocumentId;
+
+        public static (string? category, VersionStamp projectVersion) TryRead(ObjectReader reader)
```

Comment:
>TryRead [](start = 70, length = 7)

This is a weird method. I'd expect it to either take DocumentId and return DesignerInfo, or not be on this type.

Reply:
Sure, i can move it.

Reply:
Done.